### PR TITLE
Clean up labeling code (first pass)

### PR DESCRIPTION
**Pull request by wonder-sk**
- Remove QgsLabel and QgsLabelAttributes + GUI (loong time unused)
- Remove QgsPalLabeling from QgsMapRendererJob - only QgsLabelingEngineV2 is handled. QgsMapRenderer still uses QgsPalLabeling (to be sorted out later)

### CODE REVIEW

**Comment by NathanW2** [comment]
Nice.

File: doc/api_break.dox [hunk 1]
```diff
@@ -101,6 +107,7 @@ screenUpdateRequested() were removed. These members have had no effect for a num
 <li>theMinScale and theMaxScale parameters in setMinimumScale and setMaximumScale were renamed to scale</li>
 <li>The layerCrsChanged() signal was renamed to crsChanged()</li>
 <li>theError parameter in appendError() and setError() were renamed to 'error'.</li>
+<li>drawLabels() mathod as removed. It used old deprecated labeling. Replaced by labeling based on PAL library, see QgsLabelingEngineV2.</li>
```
Comment (nyalldawson) [inline doc/api_break.dox]: method

**Comment by nyalldawson** [comment]
-5000 lines of cruft? Yes please!

File: doc/api_break.dox (hunk 1 above)
Comment (wonder-sk) [inline doc/api_break.dox]: Oh... two typos in two words - a clear sign of late night coding :)

**Comment by dakcarto** [comment]
@wonder-sk, thanks for this!

**Comment by wonder-sk** [comment]
This is something I have been looking forward to do since 2009 :-)

**Pull request merged**